### PR TITLE
Add OTP Geocoder

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
OTP has a geocoder now! https://docs.opentripplanner.org/en/dev-2.x/sandbox/GeocoderAPI/

This PR lets you use it within otp-ui. Only the stops endpoint for now, as it would be a lot more work to get the rest of the layers to work (would have to parse the label strings) and they're not that high quality. ARCGIS provides better results without an API key.

### CODE REVIEW

**Review by amy-corson-ibigroup — APPROVED** [review]
Looks good to me! Very cool to read about the OTP geocoder and see how you're fitting it together here.

Just want to double check, running this locally I'm not seeing stops data come up on location field search, is that expected for this PR?

File: packages/geocoder/src/apis/otp/index.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    query: text,
+    url
+  })
+}
+
+function search(): Promise<unknown> { console.warn("Not implemented"); return null }
+function reverse(): Promise<unknown> { console.warn("Not implemented"); return null }
+
+
+export { autocomplete, reverse, search,  };
```
Comment (daniel-heppner-ibigroup) [inline packages/geocoder/src/apis/otp/index.ts:50]: I honestly support trailing commas on the last item in a list like this, but I don't think it's compliant with our code formatting standard.

File: packages/geocoder/src/apis/otp/index.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    .then(res => res.text())
+    .then(res => JSON.parse(`{"results": ${res}}`));
+}
+
+/**
+ * Search for an address using
+ * OTP Geocoder
+ *
+ * @param  {Object} $0
+ * @param  {string} $0.url                    The OTP instance, ending with /default/
```
Comment (daniel-heppner-ibigroup) [inline packages/geocoder/src/apis/otp/index.ts:32]: so many spaces!

File: packages/geocoder/src/apis/otp/index.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    id: string
+  }[]
+}
+
+
+function run({  query, url }: FetchArgs): Promise<OTPGeocoderResponse> {
+  // TODO: Support corners/osm nodes?
+  return fetch(`${url}/geocode?corners=false&query=${query}`)
+    .then(res => res.text())
+    .then(res => JSON.parse(`{"results": ${res}}`));
```
Comment (daniel-heppner-ibigroup) [inline packages/geocoder/src/apis/otp/index.ts:24]: It feels a little weird to construct a JSON object as a string. Why does the results need to be wrapped in an object under "results" anyway? It seems like in the `geocoders/otp` file you could just remove the `.results` part of accessing the data and then also remove this string.

File: packages/geocoder/src/apis/otp/index.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  results: {
+    lat: number,
+    lng: number,
+    description: string,
+    id: string
+  }[]
+}
+
+
+function run({  query, url }: FetchArgs): Promise<OTPGeocoderResponse> {
```
Comment (daniel-heppner-ibigroup) [inline packages/geocoder/src/apis/otp/index.ts:20]: Maybe we should specify in the return type that it's possible to get undefined as a response, which will help enforce null coalescing operators when the function is used.

**Review by daniel-heppner-ibigroup — CHANGES_REQUESTED** [review]
In addition to my comments, can you please add the OTP geocoder to the story to make testing easier?

**Comment by miles-grant-ibigroup** [comment]
@amy-corson-ibigroup yeah this doesn't affect the location field at all at the moment

**Comment by miles-grant-ibigroup** [comment]
@daniel-heppner-ibigroup which OTP server do you think we should point to in the story? want to add a field to set one?

File: packages/geocoder/src/apis/otp/index.ts (hunk 1 above)
Comment (miles-grant-ibigroup) [reply packages/geocoder/src/apis/otp/index.ts:50]: Not sure how this slipped in to begin with...

File: packages/geocoder/src/apis/otp/index.ts (hunk 3 above)
Comment (miles-grant-ibigroup) [reply packages/geocoder/src/apis/otp/index.ts:24]: I wish... Unfortunately the response returned by OTP is invalid JSON so we have to do some processing to it. I tried to figure out how to get a "raw" array working without pelias-stitch throwing a fit, but will have another go...

Comment (daniel-heppner-ibigroup) [reply packages/geocoder/src/apis/otp/index.ts:24]: Oh, OTP is returning a straight up array? I see. That should be an easy fix in OTP, we should consider ticketing it in OTP upstream.

Comment (miles-grant-ibigroup) [reply packages/geocoder/src/apis/otp/index.ts:24]: Others rely on this api being as is

**Comment by daniel-heppner-ibigroup** [comment]
@miles-grant-ibigroup A field for the endpoint seems fine for me. Should we include a default or is that too much?

File: packages/geocoder/src/apis/otp/index.ts (hunk 3 above)
Comment (daniel-heppner-ibigroup) [reply packages/geocoder/src/apis/otp/index.ts:24]: I think we could just leave an issue for it then, at least

**Comment by miles-grant-ibigroup** [comment]
@daniel-heppner-ibigroup I've added story support for the OTP geocoder,  let me know what you think!

**Review by daniel-heppner-ibigroup — APPROVED** [review]
The story is looks really good, loving all the fields with no padding. Thanks for the work!

**Comment by daniel-heppner-ibigroup** [comment]
I just noticed that I got a yarn.lock change when I ran yarn on this. Please double check that yarn.lock is updated

**Comment by miles-grant-ibigroup** [comment]
Good catch thanks!

**Pull request merged**